lib/api/src/ticketAPI.cpp

```diff
@@ -0,0 +1,418 @@
+#include "irods/ticketAdmin.h"
+
+namespace irods::administration::ticket
+{
+    void make_ticket_name(char* newTicket)
+    {
+        const int ticket_len = 15;
+        // random_bytes must be (unsigned char[]) to guarantee that following
+        // modulo result is positive (i.e. in [0, 61])
+        unsigned char random_bytes[ticket_len];
+        irods::getRandomBytes(random_bytes, ticket_len);
+
+        const char characterSet[] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+                                     'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+                                     'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+                                     'w', 'x', 'y', 'z', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+
+        for (int i = 0; i < ticket_len; ++i) {
+            const int ix = random_bytes[i] % sizeof(characterSet);
+            newTicket[i] = characterSet[ix];
+        }
+        newTicket[ticket_len] = '\0';
+        printf("ticket:%s\n", newTicket);
+    }
+    int ticket_manager(RxComm& conn,
+                       std::string_view command,
+                       std::string_view ticket_IDentifier,
+                       std::string_view commandModifier,
+                       std::string_view commandModifier2,
+                       std::string_view commandModifier3,
+                       std::string_view commandModifier4,
+                       bool run_as_admin)
+    {
+        if (const int status = clientLogin(&conn); status != 0) {
+            return 3; // int error code -- client login didn't work
+        }
+
+        ticketAdminInp_t ticketAdminInp{};
+
+        ticketAdminInp.arg1 = strdup(command.data());
+        ticketAdminInp.arg2 = strdup(ticket_IDentifier.data());
+        ticketAdminInp.arg3 = strdup(commandModifier.data());
+        ticketAdminInp.arg4 = strdup(commandModifier2.data());
+        ticketAdminInp.arg5 = strdup(commandModifier3.data());
+        ticketAdminInp.arg6 = strdup(commandModifier4.data());
+
+        if (run_as_admin) {
+            addKeyVal(&ticketAdminInp.condInput, ADMIN_KW, "");
+        }
+
+        const int status = rcTicketAdmin(&conn, &ticketAdminInp);
+
+        if (status < 0) {
+            if (conn.rError) {
+                rError_t* Err;
+                rErrMsg_t* ErrMsg;
+                int i, len;
+                Err = conn.rError;
+                len = Err->len;
+                for (i = 0; i < len; i++) {
+                    ErrMsg = Err->errMsg[i];
+                    rodsLog(LOG_ERROR, "Level %d: %s", i, ErrMsg->msg);
+                }
+            }
+            char* mySubName = NULL;
+            const char* myName = rodsErrorName(status, &mySubName);
+            rodsLog(LOG_ERROR, "rcTicketAdmin failed with error %d %s %s", status, myName, mySubName);
+            free(mySubName);
+        }
```

Comment:
Remove these lines.

Returning the error code is good enough.

Printing the rError information in the connection object is not this library's responsibility.